bindings/cs/MongoDB.Crypt/Library.cs

```diff
@@ -27,61 +27,129 @@ public class Library
-            mongocrypt_ctx_setopt_key_alt_name = loader.GetFunction<Delegates.mongocrypt_ctx_setopt_key_alt_name>("mongocrypt_ctx_setopt_key_alt_name");
-            mongocrypt_ctx_setopt_key_id = loader.GetFunction<Delegates.mongocrypt_ctx_setopt_key_id>("mongocrypt_ctx_setopt_key_id");
-
-            mongocrypt_ctx_setopt_algorithm = loader.GetFunction<Delegates.mongocrypt_ctx_setopt_algorithm>("mongocrypt_ctx_setopt_algorithm");
-            mongocrypt_ctx_status = loader.GetFunction<Delegates.mongocrypt_ctx_status>("mongocrypt_ctx_status");
-            mongocrypt_ctx_encrypt_init = loader.GetFunction<Delegates.mongocrypt_ctx_encrypt_init>("mongocrypt_ctx_encrypt_init");
-            mongocrypt_ctx_decrypt_init = loader.GetFunction<Delegates.mongocrypt_ctx_decrypt_init>("mongocrypt_ctx_decrypt_init");
-            mongocrypt_ctx_explicit_encrypt_init = loader.GetFunction<Delegates.mongocrypt_ctx_explicit_encrypt_init>("mongocrypt_ctx_explicit_encrypt_init");
-            mongocrypt_ctx_explicit_decrypt_init = loader.GetFunction<Delegates.mongocrypt_ctx_explicit_decrypt_init>("mongocrypt_ctx_explicit_decrypt_init");
-            mongocrypt_ctx_datakey_init = loader.GetFunction<Delegates.mongocrypt_ctx_datakey_init>("mongocrypt_ctx_datakey_init");
-            mongocrypt_ctx_state = loader.GetFunction<Delegates.mongocrypt_ctx_state>("mongocrypt_ctx_state");
-            mongocrypt_ctx_mongo_op = loader.GetFunction<Delegates.mongocrypt_ctx_mongo_op>("mongocrypt_ctx_mongo_op");
-            mongocrypt_ctx_mongo_feed = loader.GetFunction<Delegates.mongocrypt_ctx_mongo_feed>("mongocrypt_ctx_mongo_feed");
-            mongocrypt_ctx_mongo_done = loader.GetFunction<Delegates.mongocrypt_ctx_mongo_done>("mongocrypt_ctx_mongo_done");
-
-            mongocrypt_ctx_next_kms_ctx = loader.GetFunction<Delegates.mongocrypt_ctx_next_kms_ctx>("mongocrypt_ctx_next_kms_ctx");
-            mongocrypt_kms_ctx_endpoint = loader.GetFunction<Delegates.mongocrypt_kms_ctx_endpoint>("mongocrypt_kms_ctx_endpoint");
-            mongocrypt_kms_ctx_message = loader.GetFunction<Delegates.mongocrypt_kms_ctx_message>("mongocrypt_kms_ctx_message");
-            mongocrypt_kms_ctx_bytes_needed = loader.GetFunction<Delegates.mongocrypt_kms_ctx_bytes_needed>("mongocrypt_kms_ctx_bytes_needed");
-            mongocrypt_kms_ctx_feed = loader.GetFunction<Delegates.mongocrypt_kms_ctx_feed>("mongocrypt_kms_ctx_feed");
-            mongocrypt_kms_ctx_status = loader.GetFunction<Delegates.mongocrypt_kms_ctx_status>("mongocrypt_kms_ctx_status");
-            mongocrypt_ctx_kms_done = loader.GetFunction<Delegates.mongocrypt_ctx_kms_done>("mongocrypt_ctx_kms_done");
-
-            mongocrypt_ctx_finalize = loader.GetFunction<Delegates.mongocrypt_ctx_finalize>("mongocrypt_ctx_finalize");
-            mongocrypt_ctx_destroy = loader.GetFunction<Delegates.mongocrypt_ctx_destroy>("mongocrypt_ctx_destroy");
+            _mongocrypt_version = new Lazy<Delegates.mongocrypt_version>(

```

Comment:
Using `Lazy` to store the delegates as a thread-safe way to delay the exception to the point when the first wrapped library function is called . Without them, since the loader is initialized in a static constructor, the exception would be thrown whenever the static constructor is called, which IMHO would result in a poor user experience (and be harder to test)

Reply:
Move the Lazy value function to a helper function?

A call to new Lazy with an argument that is 120+ lines long is odd.

Reply:
Sorry, I'm slightly confused :) There are a lot of `Lazy` values being initialized here, but each call to lazy is at most two lines long: did you still want to move the initialization of all the variables to a helper function though? (That would prevent the fields from being `static readonly` though)

Reply:
My comment is that a call to a Lazy constructor with an inline lambda with 120+ lines is a bit much.

But if that's the only way to do it so be it.